### PR TITLE
Automatically create release notes and set bump

### PR DESCRIPTION
**Pull request by brianstrauch**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Add a "Release Notes" section to the PR template, where the PR author can declare the breaking changes / new features / bug fixes corresponding to the PR.

Then, during a release we automatically compile these release notes and use them to infer the semver update type (major, minor, patch).

This is the first step in automating the entire release (so we can release via Semaphore), removes about 5-10 minutes of work writing the release notes during each release, and removes the need to supply `export BUMP=major/minor/patch`.

https://github.com/confluentinc/cli/pull/1720 needs to be merged for a full release cycle before this PR is merged.

After this PR, the order of the release will change to:
* `make release-notes` (this is now a single step, and determines major/minor/patch, to be used by the other targets)
* `make update-whitelist`
* `make update-db`
* `make release`
* `make release-docs`

Test & Review
-------------
Manually tested all affected make targets

### CODE REVIEW

**Comment by MuweiHe** [comment]
One thing i noticed about our release notes is that we actually upload to s3 before creating/merging the pr into docs, so whatever changes we made into that pr won't be synched into s3. Do you think this pr will solve that?

**Comment by brianstrauch** [comment]
@MuweiHe No, but we should definitely do that in another PR!

The process should probably look something like this:
1. `make release-notes` opens the docs PR
2. The docs PR is reviewed
3. `make release` publishes the release notes to S3 (or GitHub, and then we can get rid our dependency on S3)

**Review by sgagniere — APPROVED** [review]
+1, just once concern

File: internal/pkg/releasenotes/release_notes.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+	for !done {
+		opts := &github.CommitsListOptions{ListOptions: github.ListOptions{Page: page}}
+		commits, _, err := client.Repositories.ListCommits(ctx, owner, repo, opts)
+		if err != nil {
+			return err
+		}
+
+		for _, commit := range commits {
+			if commit.GetSHA() == latestReleaseTagSha {
```
[the remark this answers is not in the document]
Comment (brianstrauch) [reply internal/pkg/releasenotes/release_notes.go:156]: Yep :+1:

**Pull request merged**